packages/core/src/Repo.ts

```diff
@@ -294,13 +294,14 @@ export class Repo<T = any> extends EventEmitter {
   changeSnapshot(documentId: string, fn: A.ChangeFn<T>) {
     // create a new automerge object from the current version's snapshot
     const oldDoc = this.getSnapshot(documentId) || {}
-    const doc: A.Doc<any> = A.from(oldDoc)
+
+    const doc: A.Doc<any> = A.from(getSnapshot(oldDoc))
```

Comment:
It does look a bit weird to have `this.getSnapshot` and `getSnapshot` used in sequence, and they mean very different things. The new method should be called `clone`.

Reply:
Or `deepClone`, or whatever

Reply:
👍 ab980c280d4f27873ad24a7e7811a5c9a495e192